### PR TITLE
Manage learners panel language reflects team vs individual assessment

### PR DESCRIPTION
**Pull request by nsprenkle**
**TL;DR -** Update language in "Manage Individual Learners" staff view to reflect whether we're managing an individual or team assignment

**What changed?**

- Updated render functions to determine if XBlock is configured for teams.
- **NOTE:** this can probably get simplified when future passes (dealing w/ team assignments) gather this information implicitly.
- Update templates to modify team/individual language
- Update version to `2.7.10`

image

**To do**
- [x] Update UI tests

**Developer Checklist**
- [x] Reviewed the [release process](./release_process.md)
- [x] Translations up to date
- [ ] ~JS minified, SASS compiled~
- [x] Test suites passing on Jenkins
- [x] Bumped version number in [setup.py](../setup.py) and [package.json](../package.json)

JIRA: [EDUCATOR-5028](https://openedx.atlassian.net/browse/EDUCATOR-5028)

FIY: @edx/masters-devs-gta

### CODE REVIEW

File: openassessment/templates/openassessmentblock/staff_area/oa_student_info.html [hunk 1]
```diff
@@ -7,9 +7,15 @@
 {% if submission %}
     <h2 class="title">
         <span class="label staff-info__student__report__summary" tabindex="-1">
-            {% blocktrans with learner=student_username %}
-                Viewing learner: {{ learner }}
-            {% endblocktrans %}
+            {% if not is_team_assignment %}
```
Comment (Andytr1) [inline openassessment/templates/openassessmentblock/staff_area/oa_student_info.html:10]: Trivial nit(for future reference): I prefer to have positive if statements unless absolutely necessary (to save my brain extra processing power). I.e: 
```
if is_team_assignment:
...
else:
```

File: openassessment/xblock/staff_area_mixin.py [hunk 2]
```diff
@@ -140,6 +140,9 @@ def get_staff_path_and_context(self):
                 self.get_staff_assessment_statistics_context(student_item["course_id"], student_item["item_id"])
             )
 
+        # Include whether or not this is a team assignment
+        context['is_team_assignment'] = self.teams_enabled
```
Comment (Andytr1) [inline openassessment/xblock/staff_area_mixin.py:144]: There is a better flag in team_mixin called `is_team_assignment` that should be used here.

File: openassessment/xblock/staff_area_mixin.py [hunk 3]
```diff
@@ -283,6 +286,7 @@ def get_student_submission_context(self, student_username, submission):
             'user_timezone': user_preferences['user_timezone'],
             'user_language': user_preferences['user_language'],
             "prompts_type": self.prompts_type,
+            "is_team_assignment": self.teams_enabled,
```
Comment (Andytr1) [inline openassessment/xblock/staff_area_mixin.py:289]: same note about `is_team_assignment_here` .

File: openassessment/templates/openassessmentblock/staff_area/oa_student_info.html (hunk 1 above)
Comment (nsprenkle) [reply openassessment/templates/openassessmentblock/staff_area/oa_student_info.html:10]: That's fair. I did this was mostly for flow: given that an individual assignment is overwhelmingly the default case, I want that to show first.

**Comment by jansenk** [comment]
me opening "files changed" in a PR with translation changes: 
image

**Comment by nsprenkle** [comment]
> me opening "files changed" in a PR with translation changes:
> image

I feel you

File: openassessment/xblock/test/test_staff_area.py [hunk 4]
```diff
@@ -746,7 +747,7 @@ def mock_formatter(entries):
         )
 
         # Enable teams
-        xblock.teams_enabled = True
+        is_team_assignment_patch.return_value = True
```
Comment (jansenk) [inline openassessment/xblock/test/test_staff_area.py:750]: If you wait until I merge my PR, you could rebase and use the `_setup_xblock_and_create_submission` function to simplify a bunch of this (https://github.com/edx/edx-ora2/pull/1412)

Comment (nsprenkle) [reply openassessment/xblock/test/test_staff_area.py:750]: I could also use `get_team_for_anonymous_user` instead of duplicating that work. Will wait until after #1412

Comment (jansenk) [reply openassessment/xblock/test/test_staff_area.py:750]: Merged!

File: openassessment/xblock/static/js/spec/lms/oa_staff_area.js [hunk 5]
```diff
@@ -435,6 +435,55 @@ describe('OpenAssessment.StaffAreaView', function() {
...
         });
 
+        it('defaults language when managing an individual assignment', function() {
+            var staffArea = createStaffArea({}, 'oa_staff_area.html'),
+                $staffToolsButton = $('.button-staff-tools', staffArea.element),
+                $staffToolsPanel = $('.wrapper--staff-tools', staffArea.element),
+                $staffToolsHeader = $('.staff-info__title', $staffToolsPanel),
+                $reportHeader,
+                $sectionHeaders;
+
```
[the remark this answers is not in the document]
Comment (nsprenkle) [reply openassessment/xblock/static/js/spec/lms/oa_staff_area.js:445]: Which part? The `$` prefix, bashing them all at the top of the function, or something else?

File: openassessment/xblock/staff_area_mixin.py [hunk 6]
```diff
@@ -213,8 +216,8 @@ def render_staff_grade_form(self, data, suffix=''):  # pylint: disable=W0613
                         self.get_username(anonymous_student_id), submission
                     )
                     # Add team info to context
-                    submission_context['teams_enabled'] = self.teams_enabled
-                    if self.teams_enabled:
+                    submission_context['teams_enabled'] = self.is_team_assignment()
```
[the remark this answers is not in the document]
Comment (nsprenkle) [reply openassessment/xblock/staff_area_mixin.py]: Have reverted. It's in an unrelated area of the code and I shouldn't have expanded the scope of the changes.

File: openassessment/xblock/staff_area_mixin.py [hunk 7]
```diff
@@ -351,6 +355,14 @@ def get_student_info_path_and_context(self, student_username):
         if submission_uuid:
             self.add_submission_context(submission_uuid, context)
 
+        # Add team info to context
+        # Note - this can probably be simplified when we get context direct from team assignments
+        context['team_name'] = None
+        if anonymous_user_id and self.is_team_assignment():
+            user = self.get_real_user(anonymous_user_id)
+            team = self.teams_service.get_team(user, self.course_id, self.selected_teamset_id)
+            context['team_name'] = getattr(team, 'name', None)
```
[the remark this answers is not in the document]
Comment (nsprenkle) [reply openassessment/xblock/staff_area_mixin.py]: It's possible for `teams_service.get_team()` to return `None` in which case `team.name` throws an `AttributeError` which is what this is meant to cover.

File: openassessment/xblock/static/js/spec/lms/oa_staff_area.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (jansenk) [reply openassessment/xblock/static/js/spec/lms/oa_staff_area.js:445]: The $ prefix

File: openassessment/xblock/staff_area_mixin.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (jansenk) [reply openassessment/xblock/staff_area_mixin.py]: neat, cool 👍
